<a href="https://colab.research.google.com/github/natbelo/C-digos-de-Estudo/blob/main/Grau_de_escolaridade_e_n%C3%ADvel_de_renda.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [34]:
from datascience import *

In [35]:
import numpy as np


In [36]:
full_table = Table.read_table('/content/drive/MyDrive/Cloud Skills Challenge - Microsoft/ca-educational-attainment-personal-income-2008-2014.csv')

In [37]:
full_table

Year                   | Age      | Gender | Educational Attainment              | Personal Income    | Population Count
01/01/2008 12:00:00 AM | 00 to 17 | Male   | Children under 15                   | No Income          | nan
01/01/2008 12:00:00 AM | 00 to 17 | Male   | No high school diploma              | No Income          | 650889
01/01/2008 12:00:00 AM | 00 to 17 | Male   | No high school diploma              | $5,000 to $9,999   | 30152
01/01/2008 12:00:00 AM | 00 to 17 | Male   | No high school diploma              | $10,000 to $14,999 | 7092
01/01/2008 12:00:00 AM | 00 to 17 | Male   | No high school diploma              | $15,000 to $24,999 | 3974
01/01/2008 12:00:00 AM | 00 to 17 | Male   | No high school diploma              | $25,000 to $34,999 | 2606
01/01/2008 12:00:00 AM | 00 to 17 | Male   | No high school diploma              | $35,000 to $49,999 | 2227
01/01/2008 12:00:00 AM | 00 to 17 | Male   | High school or equivalent           | No Income          | nan
01/01/2008 12:00:00 AM | 00 to 17 | Male   | Some college, less than 4-yr degree | No Income          | 8664
01/01/2008 12:00:00 AM | 00 to 17 | Male   | Some college, less than 4-yr degree | $10,000 to $14,999 | 1304
... (1050 rows omitted)

In [43]:
ca_2014 = full_table.where('Year', are.equal_to('01/01/2014 12:00:00 AM')).where('Age', are.not_equal_to('00 to 17'))
ca_2014

Year                   | Age      | Gender | Educational Attainment    | Personal Income    | Population Count
01/01/2014 12:00:00 AM | 18 to 64 | Male   | No high school diploma    | No Income          | 376783
01/01/2014 12:00:00 AM | 18 to 64 | Male   | No high school diploma    | $5,000 to $9,999   | 158572
01/01/2014 12:00:00 AM | 18 to 64 | Male   | No high school diploma    | $10,000 to $14,999 | 231957
01/01/2014 12:00:00 AM | 18 to 64 | Male   | No high school diploma    | $15,000 to $24,999 | 460129
01/01/2014 12:00:00 AM | 18 to 64 | Male   | No high school diploma    | $25,000 to $34,999 | 288460
01/01/2014 12:00:00 AM | 18 to 64 | Male   | No high school diploma    | $35,000 to $49,999 | 194888
01/01/2014 12:00:00 AM | 18 to 64 | Male   | No high school diploma    | $50,000 to $74,999 | 104646
01/01/2014 12:00:00 AM | 18 to 64 | Male   | No high school diploma    | $75,000 and over   | 54734
01/01/2014 12:00:00 AM | 18 to 64 | Male   | High school or equivalent | No Income          | 405204
01/01/2014 12:00:00 AM | 18 to 64 | Male   | High school or equivalent | $5,000 to $9,999   | 238754
... (117 rows omitted)

In [44]:
educ_inc = ca_2014.select('Educational Attainment','Personal Income','Population Count')
educ_inc

Educational Attainment    | Personal Income    | Population Count
No high school diploma    | No Income          | 376783
No high school diploma    | $5,000 to $9,999   | 158572
No high school diploma    | $10,000 to $14,999 | 231957
No high school diploma    | $15,000 to $24,999 | 460129
No high school diploma    | $25,000 to $34,999 | 288460
No high school diploma    | $35,000 to $49,999 | 194888
No high school diploma    | $50,000 to $74,999 | 104646
No high school diploma    | $75,000 and over   | 54734
High school or equivalent | No Income          | 405204
High school or equivalent | $5,000 to $9,999   | 238754
... (117 rows omitted)

In [46]:
education = educ_inc.select('Educational Attainment', 'Population Count')
educ_totals = education.group('Educational Attainment', sum)
educ_totals

Educational Attainment              | Population Count sum
Bachelor's degree or higher         | 8.5257e+06
High school or equivalent           | 6.29414e+06
No high school diploma              | 4.25828e+06
Some college, less than 4-yr degree | 7.7755e+06

As contagens são tão grandes que é mais prático observar os percentuais. Para isso, usaremos a função percents

In [48]:
def percents(array_x):
  return np.round((array_x/sum(array_x))*100,2)

In [49]:
educ_distribution = educ_totals.with_column(
    'Population Percent', percents(educ_totals.column(1))
)
educ_distribution

Educational Attainment              | Population Count sum | Population Percent
Bachelor's degree or higher         | 8.5257e+06           | 31.75
High school or equivalent           | 6.29414e+06          | 23.44
No high school diploma              | 4.25828e+06          | 15.86
Some college, less than 4-yr degree | 7.7755e+06           | 28.96

Usando pivot, podemos obter uma tabela de contagens de adulto californianos com classificação cruzada por Educational Attainment e Personal Income.

In [50]:
totals = educ_inc.pivot('Educational Attainment','Personal Income', values = 'Population Count', collect = sum)
totals
#Cada coluna de contagens é uma distribuição do nível de renda pessoal em um grau de escolaridade específico

Personal Income    | Bachelor's degree or higher | High school or equivalent | No high school diploma | Some college, less than 4-yr degree
$10,000 to $14,999 | 452449                      | 692661                    | 664607                 | 798596
$15,000 to $24,999 | 773684                      | 1.25238e+06               | 875498                 | 1.34526e+06
$25,000 to $34,999 | 693884                      | 929218                    | 464564                 | 1.09164e+06
$35,000 to $49,999 | 1.12279e+06                 | 782804                    | 260579                 | 1.11242e+06
$5,000 to $9,999   | 326020                      | 626499                    | 597039                 | 810641
$50,000 to $74,999 | 1.59468e+06                 | 525517                    | 132516                 | 883826
$75,000 and over   | 2.9867e+06                  | 323192                    | 58945                  | 748103
No Income          | 575491                      | 1.16187e+06               | 1.20453e+06            | 985011

In [51]:
distributions = totals.select(0).with_columns(
    "Bachelor's degree or higher", percents(totals.column(1)),
    'College, less than 4-yr degree', percents(totals.column(2)),
    'High school or equivalent', percents(totals.column(3)),
    'No high school diploma', percents(totals.column(4))   
    )

distributions

Personal Income    | Bachelor's degree or higher | College, less than 4-yr degree | High school or equivalent | No high school diploma
$10,000 to $14,999 | 5.31                        | 11                             | 15.61                     | 10.27
$15,000 to $24,999 | 9.07                        | 19.9                           | 20.56                     | 17.3
$25,000 to $34,999 | 8.14                        | 14.76                          | 10.91                     | 14.04
$35,000 to $49,999 | 13.17                       | 12.44                          | 6.12                      | 14.31
$5,000 to $9,999   | 3.82                        | 9.95                           | 14.02                     | 10.43
$50,000 to $74,999 | 18.7                        | 8.35                           | 3.11                      | 11.37
$75,000 and over   | 35.03                       | 5.13                           | 1.38                      | 9.62
No Income          | 6.75                        | 18.46                          | 28.29                     | 12.67

Em uma visão geral, você pode notar que mais de 35% das pessoas com diploma de bacharel ou superior alcançaram uma renda de US$ 75.000 ou mais, ao passo que menos de 10% das pessoas nos demais graus de escolaridade alcançaram esse nível de renda.